rte/kernels/mo_fluxes_broadband_kernels.F90

```diff
@@ -33,22 +33,21 @@ pure subroutine sum_broadband(ncol, nlev, ngpt, spectral_flux, broadband_flux) b
     real(wp), dimension(ncol, nlev, ngpt), intent(in ) :: spectral_flux
     real(wp), dimension(ncol, nlev),       intent(out) :: broadband_flux
 
-    integer :: icol, ilev, igpt
+    integer  :: icol, ilev, igpt
+    real(wp) :: my_bb_flux
 
     !$acc enter data copyin(spectral_flux) create(broadband_flux)
-    !$acc parallel loop collapse(2)
+    !$acc parallel loop gang vector collapse(2)
     do ilev = 1, nlev
       do icol = 1, ncol
-        broadband_flux(icol, ilev) =  spectral_flux(icol, ilev, 1)
-      end do
-    end do
-    !$acc parallel loop collapse(3)
-    do igpt = 2, ngpt
-      do ilev = 1, nlev
-        do icol = 1, ncol
-          !$acc atomic update
-          broadband_flux(icol, ilev) = broadband_flux(icol, ilev) + spectral_flux(icol, ilev, igpt)
+
+        my_bb_flux = 0.0_wp
```

Comment:
Can you explain (to me, not necessarily in the code) why we use a scalar and only update the array element at the end?

Reply:
The scalar in principle should reside in processor register (on CPU and GPU). This is the fastest memory available, and so it is useful since `my_bb_flux` is updated every iteration of the inner loop.
However, the compilers would likely do such an optimization anyway, I just wanted to make it totally explicit